public/js/controllers/charts_controller.js

```diff
@@ -605,9 +669,20 @@ export default class extends Controller {
   }
 
   _zoomCallback (start, end) {
+    let currentSettings = {}
```

Comment:
This is not the place to be changing arbitrary information in the `settings` object. If you need to update the settings for a new chart, it should be done in `plotGraph`.